comp110-worksheet-7/DirectoryUtils.cs

```diff
@@ -7,54 +7,162 @@
+                count += file.filesize;
+            }
+            Console.WriteLine("TOTAL FILE SIZE:: {0}", count);
+            return count;
+        }
+
+        // Return the number of files (not counting directories) below the given directory
+        public static int CountFiles(string directory)
+        {
+            SetFileInformation(directory);
+            return fileinfos.fileList.Count;
+        }
+
+        // Return the nesting depth of the given directory. A directory containing only files (no subdirectories) has a depth of 0.
+        public static int GetDepth(string directory)
+        {
+            string[] subDirectories = Directory.GetDirectories(directory);
+            foreach (string subdir in subDirectories)
+            {
+                GetDepth(subdir);
+                depthCount++;
+            }
+
+            return depthCount;
+            //throw new NotImplementedException();
+        }
+
+        // Get the path and size (in bytes) of the smallest file below the given directory
+        public static Tuple<string, long> GetSmallestFile(string directory)
+        {
+            SetFileInformation(directory);
+            Tuple<string, long> smallestFile = new Tuple<string, long>("", 10000);
+            foreach (FileInformation file in fileinfos.fileList)
+            {
+                if (file.filesize < smallestFile.Item2)
+                {
+                    smallestFile = new Tuple<string, long>(file.filepath, file.filesize);
+                }
+            }
+
+            return smallestFile;
+        }
+
+        // Get the path and size (in bytes) of the largest file below the given directory
+        public static Tuple<string, long> GetLargestFile(string directory)
+        {
+            SetFileInformation(directory);
+            Tuple<string, long> largestFile = new Tuple<string, long>("", 0);
+            foreach (FileInformation file in fileinfos.fileList)
+            {
+                if (file.filesize > largestFile.Item2)
+                {
+                    largestFile = new Tuple<string, long>(file.filepath, file.filesize);
+                }
+            }
+            return largestFile;
+        }
+
+        // Get all files whose size is equal to the given value (in bytes) below the given directory
+        public static IEnumerable<string> GetFilesOfSize(string directory, long size)
+        {
+            SetFileInformation(directory);
+            List<string> matchingFileSizes = new List<string>();
+
+            foreach (FileInformation file in fileinfos.fileList)
+            {
+                if (file.filesize == size && !matchingFileSizes.Contains(file.filepath))
+                {
+                    matchingFileSizes.Add(file.filepath);
+                }
+            }
+
+            return matchingFileSizes.ToArray();
+        }
+
+    }
+
+    // Stores the path and size of a file.
+    public class FileInformation
```

Comment:
It's good practice to keep each class definition in a separate .cs file, named the same as the class. In fact Unity enforces this with MonoBehaviour classes so it's a good habit to get into.